uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -124,6 +124,15 @@ impl Type {
     pub fn ffi_type(&self) -> FFIType {
         self.into()
     }
+
+    pub fn iter_types(&self) -> TypeIterator<'_> {
+        let nested_types = match self {
+            Type::Optional(t) | Type::Sequence(t) => t.iter_types(),
+            Type::Map(k, v) => Box::new(k.iter_types().chain(v.iter_types())),
+            _ => Box::new(iter::empty()),
+        };
+        Box::new(std::iter::once(self).chain(nested_types))
+    }
```

Comment:
This isn't new, but I'll just note that I find it a bit confusing how this function is split off from the `RecursiveTypeIterator` code and it works differently.  For example, `RecursiveTypeIterator` does work to prevent infinite loops, but this doesn't.  I think the code is correct, because `Optional`, `Sequence`, and `Map` can't have circular references by their nature, but it took me a second to realize that.  This is part of a bigger issue and not related to these changes.  I made #1254 to work on this further.